creator-node/src/snapbackSM.js

```diff
@@ -348,60 +362,93 @@ class SnapbackSM {
         }
       )
     )
-    let previousModuloSlice = this.currentModuloSlice
+
     // Increment and adjust current slice by ModuloBase
+    const previousModuloSlice = this.currentModuloSlice
     this.currentModuloSlice += 1
     this.currentModuloSlice = this.currentModuloSlice % ModuloBase
-    this.log(`Updated modulo slice from ${previousModuloSlice} to ${this.currentModuloSlice}`)
+    this.log(`Updated modulo slice from ${previousModuloSlice} to ${this.currentModuloSlice} out of ${ModuloBase}`)
+
     this.log(`Issued ${numSyncsIssued} sync ops`)
-    this.log(`------------------END Process SnapbackSM Operation, slice ${previousModuloSlice} ------------------`)
+    this.log(`------------------END Process SnapbackSM Operation, slice ${previousModuloSlice} / ${ModuloBase} ------------------`)
   }
 
-  // Track an ongoing sync operation to a secondary
-  async monitorSecondarySync (syncWallet, primaryClockValue, secondaryUrl) {
-    let startTime = Date.now()
-    // Monitor the sync status
-    let syncMonitoringRequestParameters = {
+  /**
+   * Track an ongoing sync operation for a given secondaryUrl and user wallet
+   * Poll repeatedly until secondary has synced up to given primaryClockValue or max duration has been exceeded
+   */
+  async monitorSecondarySync (userWallet, primaryClockValue, secondaryUrl) {
+    const startTime = Date.now()
+
+    // Define axios request object for secondary sync status request
+    const syncMonitoringRequestParameters = {
       method: 'get',
       baseURL: secondaryUrl,
-      url: `/sync_status/${syncWallet}`,
+      url: `/sync_status/${userWallet}`,
       responseType: 'json'
     }
+
+    // TODO syncAttemptCompleted does nothing - refactor this
     let syncAttemptCompleted = false
+    let secondaryClockValAfterSync = null
     while (!syncAttemptCompleted) {
       try {
-        let syncMonitoringResp = await axios(syncMonitoringRequestParameters)
-        let respData = syncMonitoringResp.data.data
-        this.log(`processSync ${syncWallet} secondary response: ${JSON.stringify(respData)}`)
+        const syncMonitoringResp = await axios(syncMonitoringRequestParameters)
+
+        const respData = syncMonitoringResp.data.data
+        this.log(`processSync ${userWallet} secondary response: ${JSON.stringify(respData)}`)
+
         // A success response does not necessarily guarantee completion, validate response data to confirm
         // Returned secondaryClockValue can be greater than the cached primaryClockValue if a client write was initiated
         //    after primaryClockValue cached and resulting sync is monitored
         if (respData.clockValue >= primaryClockValue) {
```

Comment:
right now this checks if the response clock >= value on the primary so if the secondary were catching up for the first time and the number of clock values > 10k, this would have to time out in order for the while loop to exit and the second one would start. this is probably okay for now but its pretty inefficient

Reply:
yep, this flow generally needs to change and i already have a note for that